thewalrus/symplectic.py

```diff
@@ -361,3 +363,53 @@ def autonne(A, rtol=1e-05, atol=1e-08, svd_order=True):
     if svd_order:
         return (vals[n : 2 * n])[::-1], U[:, ::-1]
     return vals[n : 2 * n], U
+
+def xxpp_to_xpxp(S):
+    """Permutes the entries of the input from xxpp ordering to xpxp ordering.
+
+    Args:
+        S (array): input even dimensional square matrix or array
+
+    Returns:
+        (array): permuted matrix or array
+    """
+    shape = S.shape
+    n = shape[0]
+
+    if n % 2 != 0:
+        raise ValueError("The input array is not even dimensional")
+
+    n = n // 2
+    ind = list(np.array([[i, i + n] for i in range(n)]).flatten())
```

Comment:
wondering if there is a better way to do this without converting to numpy array, flattening and the converting to list again...

Reply:
This avoids the list comprehension:

```suggestion
    ind = list(np.arange(2 * n).reshape(2, -1).T.flatten())
```

Does it work if you keep it as a numpy array? e.g.,
```python
ind = np.arange(2 * n).reshape(2, -1).T.flatten()
```

Or does it need to be converted to a tuple/list for NumPy to index correctly?